### PR TITLE
fix(UIShell): omit passing `isSideNavExpanded` prop to Link component

### LINKED ISSUE #4348: [React]: isSidenavExpanded prop passed to Link component throws React warning
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

For the `SideNavLink` component in UIShell, a React warning is emitted to the console as an error when the `SideNav` prop `expanded` is `true`. The `isSideNavExpanded` prop in `SideNavLink` is passed to the [`Link` component](https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/UIShell/SideNavLink.js#L35). Because it is not a valid DOM attribute, React throws a warning.

This isn't a functional bug; rather, the aim is to remove the React warning.

Omitting passing the `isSideNavExpanded` prop to the `Link` component would eliminate the error.

## Steps to reproduce the issue

1. Visit Reduced Test case in [CodeSandbox](https://codesandbox.io/s/react-carbon-uishell-ulxhu)
2. Open the console
3. React warning is printed in the console as an error

## Additional information

React warning

### PR DESCRIPTION
**Pull request by metonym**
Closes #4348

Omit passing `isSideNavExpanded` prop by destructuring it in props.

#### Changelog

**Changed**

- destructure `isSideNavExpanded` prop

#### Testing / Reviewing

In the React workspace, run `yarn start` to start the storybook locally. Navigate to the [UI Shell header base w/ sidenav story](http://localhost:9000/?path=/story/ui-shell--header-base-w-sidenav). Open the console; the React warning should not be emitted.

### CODE REVIEW

File: packages/react/src/components/UIShell/SideNavLink.js [hunk 1]
```diff
@@ -16,12 +16,15 @@ import SideNavLinkText from './SideNavLinkText';
...
 
+/* eslint-disable no-unused-vars */
+
 const SideNavLink = ({
   className: customClassName,
   children,
   renderIcon: IconElement,
   isActive,
   large,
+  isSideNavExpanded,
```
Comment (joshblack) [inline packages/react/src/components/UIShell/SideNavLink.js:26]:
```suggestion
  // eslint-disable-next-line no-unused-vars
  isSideNavExpanded,
```

**Review by joshblack — CHANGES_REQUESTED** [review]
Thanks for doing this! Just curious, does this syntax end up disabling the eslint violation? Would be great if we could scope it to the line instead of the file 👍

File: packages/react/src/components/UIShell/SideNavLink.js [hunk 2]
```diff
@@ -16,12 +16,16 @@ import SideNavLinkText from './SideNavLinkText';
 
 const { prefix } = settings;
 
+/* eslint-disable no-unused-vars */
```
Comment (asudoh) [inline packages/react/src/components/UIShell/SideNavLink.js]:
```suggestion
```

Given we disable `no-unused-vars` for a specific line.

File: packages/react/src/components/UIShell/SideNavLink.js [hunk 3]
```diff
@@ -16,12 +16,15 @@ import SideNavLinkText from './SideNavLinkText';
...
 const { prefix } = settings;
 
+
 const SideNavLink = ({
   className: customClassName,
   children,
   renderIcon: IconElement,
   isActive,
   large,
+  // eslint-disable-next-line no-unused-vars
```
Comment (asudoh) [inline packages/react/src/components/UIShell/SideNavLink.js:25]: Thank you for the update; BTW I thought https://twitter.com/dan_abramov/status/838576620235550720 is applied to our ESLint config. Did you see it's not the case...?

Comment (metonym) [reply packages/react/src/components/UIShell/SideNavLink.js:25]: Before I pushed the first commit, I didn't disable any eslint rules. I ran "yarn ci-check" on my local machine and then faced the error "no-unused-vars."

Comment (asudoh) [reply packages/react/src/components/UIShell/SideNavLink.js:25]: OK, fair enough - Thank you for clarifying!

**Review by asudoh — APPROVED** [review]

**Review by joshblack — APPROVED** [review]

**Comment by asudoh** [comment]
@metonym Would you be able to re-run Prettier? Thanks!

**Comment by Debirpriya** [comment]
The UiShell SideNav behavior is fixed in 10.7.1 .
10.3.1 version the rail menu does not behave properly

**Pull request merged**